src/logic/initializer/AppInitializer.ts

```diff
@@ -4,18 +4,30 @@ import {store} from "../../index";
 import {PlatformUtil} from "../../utils/PlatformUtil";
 import {PlatformModel} from "../../staticModels/PlatformModel";
 import {EventType} from "../../data/enums/EventType";
+import {GeneralSelector} from "../../store/selectors/GeneralSelector";
 
 export class AppInitializer {
     public static inti():void {
         AppInitializer.handleResize();
         AppInitializer.detectDeviceParams();
+        AppInitializer.handleAccidentalPageExit();
         window.addEventListener(EventType.RESIZE, AppInitializer.handleResize);
         window.addEventListener(EventType.MOUSE_WHEEL, AppInitializer.disableGenericScrollZoom,{passive:false});
         window.addEventListener(EventType.KEY_DOWN, AppInitializer.disableUnwantedKeyBoardBehaviour);
         window.addEventListener(EventType.KEY_PRESS, AppInitializer.disableUnwantedKeyBoardBehaviour);
         ContextManager.init();
     }
 
+    private static handleAccidentalPageExit = () => {
+        window.onbeforeunload = (event) => {
```

Comment:
generalnie jest OK, 
ale nie wiem czy nie lepiej by było zrobić to normalnie: 
```window.addEventListener('beforeunload', ...```
sprawdź czy zadziała na wszystkich przeglądarkach. ale powinno